test/gtest/activation.cpp

```diff
@@ -0,0 +1,262 @@
+
+        dinput_gpu.data = handle.Read<float>(din_dev, dinput_gpu.data.size());
+
+        CompareTensors(dinput_cpu, dinput_gpu);
+        miopenDestroyActivationDescriptor(activ_desc);
+    }
+
+    tensor<float> input; // x
+    tensor<float> output_gpu;
+    tensor<float> output_cpu_ref; // y
+
+    tensor<float> dinput_cpu; // dx
+    tensor<float> dinput_gpu;
+    tensor<float> doutput;
+
+    ActivationConfig activ_config;
+    miopenActivationMode_t activ_mode;
+    miopenActivationDescriptor_t activ_desc;
+    miopen::Allocator::ManageDataPtr in_dev;   // x
+    miopen::Allocator::ManageDataPtr out_dev;  // y
+    miopen::Allocator::ManageDataPtr din_dev;  // dx
+    miopen::Allocator::ManageDataPtr dout_dev; // dy
+};
+
+miopenStatus_t RunActivation(miopen::Handle& handle,
+                             miopenActivationDescriptor_t activationDesc,
+                             const void* alpha,
+                             const miopen::TensorDescriptor& xDesc,
+                             ConstData_t x,
+                             const void* beta,
+                             const miopen::TensorDescriptor& yDesc,
+                             Data_t y,
+                             const miopen::TensorDescriptor& dyDesc,
+                             ConstData_t dy,
+                             const miopen::TensorDescriptor& dxDesc,
+                             Data_t dx)
+{
+    if(alpha == nullptr || beta == nullptr)
+        MIOPEN_THROW(miopenStatusBadParm, "alpha or beta is NULL");
+
+    miopen::ActivationDescriptor desc = miopen::deref(activationDesc);
+    miopenStatus_t fwdStatus          = desc.Forward(handle,
+                                            alpha,
+                                            xDesc, // input.desc
+                                            x,     // in_dev.get()
+                                            beta,
+                                            yDesc, // output_gpu.desc
+                                            y);    // out_dev.get()
+
+    miopenStatus_t bwdStatus = desc.Backward(handle,
+                                             alpha,
+                                             yDesc, // out.desc
+                                             y,     // out_dev.get()
+                                             dyDesc,
+                                             dy,
+                                             xDesc, // input.desc
+                                             x,
+                                             beta,
+                                             dxDesc,
+                                             dx);
+
+    if(fwdStatus == miopenStatusSuccess && bwdStatus == miopenStatusSuccess)
+        return miopenStatusSuccess;
+    else if(fwdStatus != miopenStatusSuccess)
+        return fwdStatus;
+    else
+        return bwdStatus;
+}
+
+INSTANTIATE_TEST_SUITE_P(ActivationTestSuite,
+                         TestActivation,
```

Comment:
Please move the instantiation to the end of the file.

Reply:
> * Is this ready for review?
Yes please
> * How can we make sure that the new test checks the same configs as the old one?
New checks take the combination of data type, activation functions/mode and configurations. For the same input, same results/coverage is expected
> * If a failure occurs with some config, what diagnostics does the test print? How to re-test the failing config in isolation?
EXPECT_ or ASSERT_ macros in the code provided by googletest would spew info when failure happens at that point